Cubical/Algebra/CommAlgebra/UnivariatePolyList.agda

```diff
@@ -2,29 +2,218 @@
 module Cubical.Algebra.CommAlgebra.UnivariatePolyList where
 
 open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Structure
 open import Cubical.Foundations.Isomorphism
 
+open import Cubical.Data.Sigma
+
 open import Cubical.Algebra.Ring
 open import Cubical.Algebra.CommRing
+open import Cubical.Algebra.AbGroup
 open import Cubical.Algebra.CommAlgebra
+open import Cubical.Algebra.Algebra
 open import Cubical.Algebra.CommRing.Instances.Polynomials.UnivariatePolyList
 open import Cubical.Algebra.Polynomials.UnivariateList.Base
 open import Cubical.Algebra.Polynomials.UnivariateList.Properties
 
+open import Cubical.Tactics.CommRingSolver.Reflection
+
 private variable
-  ℓ : Level
+  ℓ ℓ' : Level
 
 module _ (R : CommRing ℓ) where
-  open CommRingStr ⦃...⦄
+  open RingStr ⦃...⦄    -- Not CommRingStr, so it can be used together with AlgebraStr
   open PolyModTheory R
   private
     ListPoly = UnivariatePolyList R
     instance
-      _ = snd ListPoly
-      _ = snd R
+      _ = snd (CommRing→Ring ListPoly)
+      _ = snd (CommRing→Ring R)
 
   ListPolyCommAlgebra : CommAlgebra R ℓ
   ListPolyCommAlgebra =
     Iso.inv (CommAlgChar.CommAlgIso R)
             (ListPoly ,
              constantPolynomialHom R)
+
+  {- Universal Property -}
+  module _ (A : Algebra (CommRing→Ring R) ℓ') where
```

Comment:
It is a bit strange to me that we're in `Algebra.CommAlgebra` but actually consider not-necessarily commutative algebras...

Reply:
Yes, I feel with you ;-)
But: We are proving the universal property of an CommAlgebra, which happens to quantify over Algebras. An analogous example where this is completely okay, would be the universal property of the abelianisation of a group. There it feels completely natural, that the ump quantifies over something more general. I think the ump of the CommAlgebra of Polynomials should also naturally quantify over something more general (e.g. Algebras), the situation is just more complex.